v0/destinations/pinterestConversion/transform.js

```diff
@@ -0,0 +1,266 @@
+  let customPayload = constructPayload(message, CUSTOM_CONFIGS);
+
+  // if product array is present will look for the product level information
+  if (
+    properties.products &&
+    Array.isArray(properties.products) &&
+    properties.products.length > 0
+  ) {
+    const { products } = properties;
+    products.forEach(product => {
+      const prodParams = setIdPriceQuantity(product, message);
+      contentIds.push(prodParams.contentId);
+      contentArray.push(prodParams.content);
+      totalQuantity = product.quantity
+        ? totalQuantity + product.quantity
+        : totalQuantity;
+    });
+
+    if (totalQuantity === 0) {
+      /* 
+      in case any of the products inside product array does not have quantity,
+       will map the quantity of root level 
+      */
+      totalQuantity = properties.quantity;
+    }
+  } else {
+    /* 
+    for the events where product array is not present, root level id, price and
+    quantity are taken into consideration 
+    */
+    const prodParams = setIdPriceQuantity(message.properties, message);
+    contentIds.push(prodParams.contentId);
+    contentArray.push(prodParams.content);
+    totalQuantity = properties.quantity
+      ? totalQuantity + properties.quantity
+      : totalQuantity;
+  }
+  /*
+    if properties.numOfItems is not provided by the user, the total quantity of the products
+    will be sent as num_items
+  */
+  if (!isDefinedAndNotNull(customPayload.num_items)) {
+    customPayload.num_items = parseInt(totalQuantity, 10);
+  }
+  customPayload = {
+    ...customPayload,
+    content_ids: contentIds,
+    contents: contentArray
+  };
+
+  const finalTrackPayload = {
+    ...mandatoryPayload,
+    custom_data: { ...customPayload }
+  };
+  return finalTrackPayload;
+};
+const process = event => {
+  let response;
+  const { message, destination } = event;
+  const messageType = message.type.toLowerCase();
+  const destConfig = destination.Config;
+
+  if (!destConfig.advertiserId) {
+    throw new CustomError("Advertiser Id not found. Aborting", 400);
+  }
+
+  if (!message.type) {
+    throw new CustomError(
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+
+  switch (messageType) {
+    case EventType.IDENTIFY:
+    case EventType.PAGE:
+      response = identifyPageResponseBuilder(message, destination);
+      break;
+    case EventType.TRACK:
+      response = trackResponseBuilder(message, destination);
+      break;
+    default:
+      throw new CustomError(
+        `message type ${messageType} is not supported`,
+        400
+      );
+  }
+  return responseBuilderSimple(response);
+};
+const generateBatchedPaylaodForArray = events => {
+  let batchEventResponse = defaultBatchRequestConfig();
+  const batchResponseList = [];
+  const metadata = [];
+  // extracting destination from the first event in a batch
+  const { destination } = events[0];
+  // Batch event into dest batch structure
+  events.forEach(ev => {
```

Comment:
Please update the name from `ev` to `event`

Reply:
done